diff --git a/dibs.module b/dibs.module
index cbeb0c6..fddf002 100644
--- a/dibs.module
+++ b/dibs.module
@@ -772,3 +772,14 @@ function dibs_capture_transaction($api_module, $api_delta, $transaction) {
   }
   return $result;
 }
+
+/*
+ * Implements hook_views_api()
+ */
+
+function dibs_views_api() {
+  return array(
+    'api' => 3,
+    'path' => drupal_get_path('module', 'dibs') . '/views',
+  );
+}
diff --git a/views/dibs.views.inc b/views/dibs.views.inc
new file mode 100644
index 0000000..f1aecbe
--- /dev/null
+++ b/views/dibs.views.inc
@@ -0,0 +1,105 @@
+<?php
+
+/*
+ * Implements hook_views_data()
+ */
+
+function dibs_views_data() {
+  $data['dibs_transactions']['table']['group'] = t('Dibs Transactions');
+
+  $data['dibs_transactions']['table']['base'] = array(
+    'field' => 'tid',
+    'title' => t('Dibs Transactions'),
+    'help' => t('Exposed Dibs Transactions.'),
+  );
+  
+  $data['dibs_transactions']['table']['join'] = array(
+    'commerce_payment_transaction' => array(
+      'left_field' => 'remote_id',
+      'field' => 'tid',
+    ),
+  );
+
+
+  $data['dibs_transactions']['tid'] = array(
+    'title' => t('Dibs Transaction ID'),
+    'help' => t('Dibs Transaction ID from Dibs Transactions table'),
+    'relationship' => array(
+      'base' => 'commerce_payment_transaction', // The name of the table to join with.
+      'base field' => 'remote_id', // The name of the field on the joined table.
+      'handler' => 'views_handler_relationship',
+      'label' => t('Commerce relationship to Dibs Transactions'),
+      'title' => t('Commerce relationship to Dibs Transactions'),
+    ),
+  );
+
+  $data['dibs_transactions']['payment_order_id'] = array(
+    'title' => t('Payment Order ID'),
+    'help' => t('Payment Order ID from Dibs Transactions table.'),
+    'field' => array(
+      'handler' => 'views_handler_field',
+      'click sortable' => TRUE, 
+    ),
+    'sort' => array(
+      'handler' => 'views_handler_sort',
+    ),
+    'filter' => array(
+      'handler' => 'views_handler_filter_string',
+    ),
+    'argument' => array(
+      'handler' => 'views_handler_argument_string',
+    ),
+  );
+
+  $data['dibs_transactions']['payment_transaction_id'] = array(
+    'title' => t('Payment Transaction ID'),
+    'help' => t('Payment Transaction ID from Dibs Transactions table.'),
+    'field' => array(
+      'handler' => 'views_handler_field_numeric',
+      'click sortable' => TRUE,
+     ),
+    'filter' => array(
+      'handler' => 'views_handler_filter_numeric',
+    ),
+    'sort' => array(
+      'handler' => 'views_handler_sort',
+    ),
+  );
+
+
+  return $data;
+}
+
+
+
+
+/**
+ * Implements hook_default_views_alter()
+ */
+
+function commerce_payment_ui_views_default_views_alter(&$views) {
+  if (isset($views['commerce_payment_order'])) {
+    $handler =& $views['commerce_payment_order']->display['default']->handler;
+    /* Moving Field: Operations to the back */
+    $temp = $handler->display->display_options['fields']['operations'];
+    unset($handler->display->display_options['fields']['operations']);
+    /* Relationship: Dibs Transactions: Commerce relationship to Dibs Transactions */
+    $handler->display->display_options['relationships']['tid']['id'] = 'tid';
+    $handler->display->display_options['relationships']['tid']['table'] = 'dibs_transactions';
+    $handler->display->display_options['relationships']['tid']['field'] = 'tid';
+    $handler->display->display_options['relationships']['tid']['label'] = 'Commerce relationship label';
+      /* Field: Dibs Transactions: Payment Transaction ID */
+    $handler->display->display_options['fields']['payment_transaction_id']['id'] = 'payment_transaction_id';
+    $handler->display->display_options['fields']['payment_transaction_id']['table'] = 'dibs_transactions';
+    $handler->display->display_options['fields']['payment_transaction_id']['field'] = 'payment_transaction_id';
+    $handler->display->display_options['fields']['payment_transaction_id']['relationship'] = 'tid';
+    $handler->display->display_options['fields']['payment_transaction_id']['separator'] = '';
+    /* Field: Dibs Transactions: Payment Order ID */
+    $handler->display->display_options['fields']['payment_order_id']['id'] = 'payment_order_id';
+    $handler->display->display_options['fields']['payment_order_id']['table'] = 'dibs_transactions';
+    $handler->display->display_options['fields']['payment_order_id']['field'] = 'payment_order_id';
+    $handler->display->display_options['fields']['payment_order_id']['relationship'] = 'tid';
+    /* Adding Field: Operation to the end */
+    $handler->display->display_options['fields']['operations'] = $temp;
+  }
+}
